graphqlapi/src/main/java/io/stargate/graphql/web/Server.java

```diff
@@ -0,0 +1,130 @@
+import io.dropwizard.setup.Environment;
+import io.dropwizard.util.JarLocation;
+import io.stargate.auth.AuthenticationService;
+import io.stargate.core.metrics.api.Metrics;
+import io.stargate.db.Persistence;
+import io.stargate.graphql.GraphqlActivator;
+import io.stargate.graphql.web.resources.GraphqlDdlResource;
+import io.stargate.graphql.web.resources.GraphqlDmlResource;
+import io.stargate.graphql.web.resources.PlaygroundResource;
+import java.util.EnumSet;
+import javax.servlet.DispatcherType;
+import javax.servlet.FilterRegistration;
+import org.eclipse.jetty.servlets.CrossOriginFilter;
+import org.glassfish.hk2.utilities.binding.AbstractBinder;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.FrameworkUtil;
+
+public class Server extends Application<ApplicationConfiguration> {
+
+  private final Persistence persistence;
+  private final AuthenticationService authenticationService;
+  private final Metrics metrics;
+  private volatile org.eclipse.jetty.server.Server jettyServer;
+
+  public Server(
+      Persistence persistence, AuthenticationService authenticationService, Metrics metrics) {
+    this.persistence = persistence;
+    this.authenticationService = authenticationService;
+    this.metrics = metrics;
+  }
+
+  /**
+   * The only reason we override this is to remove the call to {@code bootstrap.registerMetrics()}.
+   *
+   * <p>JVM metrics are registered once at the top level in the health-checker module.
+   */
+  @Override
+  public void run(String... arguments) {
+    final Bootstrap<ApplicationConfiguration> bootstrap = new Bootstrap<>(this);
+    addDefaultCommands(bootstrap);
+    initialize(bootstrap);
+
+    final Cli cli = new Cli(new JarLocation(getClass()), bootstrap, System.out, System.err);
+    // only exit if there's an error running the command
+    cli.run(arguments).ifPresent(this::onFatalError);
+  }
+
+  @Override
+  public void run(
+      final ApplicationConfiguration applicationConfiguration, final Environment environment) {
+
+    GraphqlCache graphqlCache = new GraphqlCache(persistence, authenticationService);
+    environment
+        .jersey()
+        .register(
+            new AbstractBinder() {
+              @Override
+              protected void configure() {
+                bind(graphqlCache).to(GraphqlCache.class);
+              }
+            });
+    environment
+        .jersey()
+        .register(
+            new AbstractBinder() {
+              @Override
+              protected void configure() {
+                bind(FrameworkUtil.getBundle(GraphqlActivator.class)).to(Bundle.class);
+              }
+            });
+
+    environment.jersey().register(GraphqlDmlResource.class);
+    environment.jersey().register(GraphqlDdlResource.class);
+    environment.jersey().register(PlaygroundResource.class);
+
+    enableCors(environment);
+
+    environment.lifecycle().addServerLifecycleListener(server -> Server.this.jettyServer = server);
```

Comment:
Why aren't we just letting dropwizard handle everything?

Reply:
I didn't find any programmatic "stop" method in the DropWizard API (I think because most of the time it's used in a dedicated VM so you just kill it).

Reply:
Got it.